File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        for(int i = Math.min(n,m);i>=1;i--){
            if(n%i==0&&m%i==0){
                answer[0] = i;
                break;
            }
        }
        int max = Math.max(n,m);
        for(long i = max;i>0;i+=max){
            if(i%n==0&&i%m==0){
                answer[1] = (int)i;
                break;
            }
        }
        return answer;
    }
}

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cc
#include <iostream>
using namespace std;

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);
	
	int a, b; cin >> a >> b;
	int mx = max(a, b);
	int mn = min(a, b);
	for (int i = mn; i > 0; i--) {
		if (a % i == 0 && b % i == 0) {
			cout << i << '\n'; break;
		}
	}
	for (int i = mx;; i += mx) {
		if (i % a == 0 && i % b == 0) {
			cout << i << '\n'; break;
		}
	}
}